### PR TITLE
Issue 7 CTA Auto block

### LINKED ISSUE #7: Contact Us CTA
**Contact us CTA**
merative-contact-us

### PR DESCRIPTION
**Pull request by amol-anand**
Fix #7 

Test URLs:
- Before: https://main--adobe-franklin-demo--amol-anand.hlx.page/
- After: https://issue-7-cta--adobe-franklin-demo--amol-anand.hlx.page/

### CODE REVIEW

**Review by rofe — APPROVED** [review]
LGTM. You may want to consider further isolation of the behavior, e.g. to _just_ an H2 and a link in a section, in order to avoid potential surprises for authors.

File: scripts/scripts.js [hunk 1]
```diff
@@ -27,13 +27,28 @@ function buildHeroBlock(main) {
...
 
+function buildCtaBlock(main) {
+  main.querySelectorAll(':scope > div').forEach((div) => {
+    const h2 = div.querySelector(':scope > h2');
+    const p = div.querySelector(':scope > p');
+    const a = div.querySelector(':scope p > a');
+    // eslint-disable-next-line no-bitwise
+    if (h2 && p && a && (h2.compareDocumentPosition(p) & Node.DOCUMENT_POSITION_FOLLOWING)
+          // eslint-disable-next-line no-bitwise
+          && (p.compareDocumentPosition(a) & Node.DOCUMENT_POSITION_CONTAINED_BY)) {
```
Comment (shsteimer) [inline scripts/scripts.js]: I think this check can be avoided by slightly changing your selectors. to `a = p.querySelector('a')` thus ensuring the link is inside the p tag

**Review by shsteimer — APPROVED** [review]
Looks fine to me.

only other note: the scripts.js already has a decorateButtons which basically makes any link on a line by itself a button, so you might consider how these 2 things interact

File: scripts/scripts.js (hunk 1 above)
Comment (amol-anand) [reply scripts/scripts.js]: Yes, the decorateButtons function basically adds a class to the p tag parent of the a link. Doesn't do anything outside of that. I am using that for my css though and my CSS is targeting that button only if it is under the CTA section as well:
`main .section.cta`

I did update the check to remove the 5th check and instead go with p.querySelector('a') and check for the existence of p before moving to the larger check. code updated and tested.

**Pull request merged**